pkgs/desktops/gnome/extensions/buildGnomeExtension.nix

```diff
@@ -36,7 +36,12 @@ let
         echo "${metadata}" | base64 --decode > $out/metadata.json
       '';
     };
-    dontBuild = true;
+    nativeBuildInputs = with pkgs; [ glib ];
+    buildPhase = ''
```

Comment:
It would be good to put `runHook preBuild` in the beginning of `buildPhase` and `runHook postBuild` in the end of `buildPhase` for easy overriding.